src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiLabelTests/MauiLabelTests.cs

```diff
@@ -0,0 +1,204 @@
+            OldSize = lbl.Size;
+            lbl.Text = "Say Hello";
+            lbl.AutoSize = true;
+            NewSize = lbl.Size;
+            return new ScenarioResult(!(NewSize.Equals(OldSize)));
+        }
+
+        [Scenario(true)]
+        public ScenarioResult AutoTFScenario(TParams p)
+        {
+            p.log.WriteLine("Make Sure autosize works on T-F toggel");
+            lbl.AutoSize = true;
+            Size initialSize = new Size(10, 10);
+            lbl.Size = initialSize;
+            lbl.Text = "Say Hello to me";
+            OldSize = lbl.Size;
+            lbl.AutoSize = false;
+            NewSize = lbl.Size;
+
+            string descr = "";
+            if (NewSize.Equals(OldSize))
+            {
+                descr = "Bug # 43661";
+            }
+
+            return new ScenarioResult((!NewSize.Equals(OldSize)) && NewSize.Equals(initialSize), descr, p.log);
+        }
+
+        [Scenario(true)]
+        public ScenarioResult AutoFFScenario(TParams p)
+        {
+            p.log.WriteLine("Make Sure autosize works on F-F toggel");
+            lbl.AutoSize = false;
+            lbl.Size = new Size(10, 10);
+            OldSize = lbl.Size;
+            lbl.Text = "is there a bug in this code here";
+            lbl.AutoSize = false;
+            NewSize = lbl.Size;
+            return new ScenarioResult(NewSize.Equals(OldSize));
+        }
+
+        [Scenario(true)]
+        public ScenarioResult AutoFTScenario(TParams p)
+        {
+            p.log.WriteLine("Make Sure autosize works on F-T toggel");
+            lbl.AutoSize = false;
+            lbl.Size = new Size(10, 10);
+            lbl.Text = "Say Hello";
+            OldSize = lbl.Size;
+            lbl.AutoSize = true;
+            NewSize = lbl.Size;
+            return new ScenarioResult(!NewSize.Equals(OldSize));
+        }
+
+        [Scenario(true)]
+        public ScenarioResult TextAlignEnumScenario(TParams p)
+        {
+            p.log.WriteLine("Make Sure I can set all Align Enums");
+            lbl.AutoSize = true;
+            try
+            {
+                lbl.TextAlign = ContentAlignment.TopLeft;
+                lbl.TextAlign = ContentAlignment.TopCenter;
+                lbl.TextAlign = ContentAlignment.TopRight;
+
+                lbl.TextAlign = ContentAlignment.MiddleLeft;
+                lbl.TextAlign = ContentAlignment.MiddleCenter;
+                lbl.TextAlign = ContentAlignment.MiddleRight;
+
+                lbl.TextAlign = ContentAlignment.BottomLeft;
+                lbl.TextAlign = ContentAlignment.BottomCenter;
+                lbl.TextAlign = ContentAlignment.BottomRight;
+            }
+            catch (Win32Exception)
+            {
+                return new ScenarioResult(false, "Failed to set all Alignment enums");
+            }
+            return ScenarioResult.Pass;
+        }
+
+        [Scenario(true)]
+        public ScenarioResult TextAlignValidLiteralScenario(TParams p)
```

Comment:
I don't think the name is correctly reflect the nature of the test.
Literals would be "TopLeft", "MiddleCenter", whereas here we test against integral values.
```suggestion
        public ScenarioResult TextAlignValidIntegralScenario(TParams p)
```

Reply:
If we change this scenario as you suggest above, this name may not need to be modified or may be there is a more appropriate name, do you think?

Reply:
We can test both integral and literal scenarios:
1. `Enum.GetValues(typeof(ContentAlignment))` gives us integral values, and
2. `Enum.GetNames(typeof(ContentAlignment))` gives us names

Reply:
Agree with Igor here, but I'd go one further and say the second scenario is not needed. All you're doing here is casting the enum value to the enum name, which is not testing winforms code. I think the scenario that tests Enum.GetNames is enough.

@RussKie what do you think?

Reply:
If we don't expect consumers to set values via literals, then we can only test integral part.
We can always add more tests, if/when it becomes necessary :)